### ISSUE TITLE
What about close dynamic modal from slot components? 

### ISSUE THREAD

**Issue by finnpham** [vue-final/vue-final-modal#56]
### Version
vue-final-modal: latest
vue: v2
nuxt: latest

### Steps to reproduce

- I have created 1 base-component like img 
![image](https://user-images.githubusercontent.com/76545073/110285092-5560ae00-8015-11eb-8ede-c929d4170b64.png)

- and created slot component img 
![image](https://user-images.githubusercontent.com/76545073/110285211-7c1ee480-8015-11eb-8ed5-8f99cba65215.png)
=> In here i have close button. **But**: I don't know how i can trigger close on base-modal to close it

- in component need use: 
![image](https://user-images.githubusercontent.com/76545073/110285347-b7211800-8015-11eb-9f30-7cb3b5236e78.png)

### What is Expected?
- I want close Modal by trigger event in `child-component` to `base-component`

### What is actually happening?
- I don't know how it can work

Thanks you for suppport me

**Comment by finnpham** [comment]
@hunterliu1003 do you have any ideal?  my opinion is pass slot function and trigger it from another component

**Comment by finnpham** [comment]
oh Thanks you @hunterliu1003 , i forgot you have provide a function for each modal have id. thanks you so much

**Comment by ddiazepam** [comment]
I kept getting "this is undefined error"
Who else expirence this should use:
`              cancel: () => {
                this.$vfm.hide(cardID)
              }`
Also "this.$vfm.hide(cardID)" could be replaced with "this.$vfm.hideAll()" if you want to close all modals

**Comment by hunterliu1003** [comment]
Do you try not to use arrow function but use `cancel() { this.$vfm.hide(cardID) }`?

**Comment by ddiazepam** [comment]
> Do you try not to use arrow function but use `cancel() { this.$vfm.hide(cardID) }`?

Yes I did. The only difference is that I use it as my own plugin, like:

`import Modal from "@/Parts/Modal";
import RepeatModal from "@/Parts/RepeatModal";
export default {
  install: (Vue, options) => {
    Vue.mixin({
      methods: {
        repeatModal(session) {
          this.$vfm.show({
            component: AppModal,
            slots: {
              default: {
                component: RepeatModal,
                bind: {
                    session: session,
                },
                on: {
                  close: () => {
                    this.$vfm.hideAll();
                  },
                },
              },
            },
          });
        },
      },
    });
  },
};
`

**Comment by hunterliu1003** [comment]
if you using vue-final-modal@2.4.x, try import $vfm directly like:



```js
import { $vfm } from 'vue-final-modal'
import Modal from "@/Parts/Modal";
import RepeatModal from "@/Parts/RepeatModal";
export default {
  install: (Vue, options) => {
    Vue.mixin({
      methods: {
        repeatModal(session) {
          $vfm.show({
            component: Modal,
            slots: {
              default: {
                component: RepeatModal,
                bind: {
                    session: session,
                },
                on: {
                  close: () => {
                    $vfm.hideAll();
                  },
                },
              },
            },
          });
        },
      },
    });
  },
};
```

**Comment by ddiazepam** [comment]
@hunterliu1003 I'm using 3.4.0 and vue3
Anyway thanks - for now it works
